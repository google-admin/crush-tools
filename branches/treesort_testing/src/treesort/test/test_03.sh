test_number=03
description="all column sort"

subtest_desc=("AA,BB,CC,DD"
							"BB,AA,CC,DD"
							"DD,AA,CC,BB"
							"3,2,4,1 indexes"
							'AA,^BB,CC,^DD'
							'AA,^BB,^CC,DD'
							'^3,1,^4,2 indexes'
							'^3,^4,^2,^1 indexes'
							)
subtest_opts=(
'-K AA,BB,CC,DD'
'-K BB,AA,CC,DD'
'-K DD,AA,CC,BB'
'-k 3,2,4,1 -p'
'-K AA,^BB,CC,^DD'
'-K AA,^BB,^CC,DD'
'-k ^3,1,^4,2 -p'
'-k ^3,^4,^2,^1 -p'
)

for subtest in `seq 0 7`; do
  expected="$test_dir/test_$test_number.$subtest.expected"
  outfile="$test_dir/test_$test_number.$subtest.actual"

  $bin ${subtest_opts[$subtest]} \
       "$test_dir/test2.in" \
       > "$outfile"

  if [ $? -ne 0 ] ||
     [ "`diff -q $outfile $expected`" ]; then
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" FAIL
  else
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" PASS
    rm "$outfile"
  fi
done
