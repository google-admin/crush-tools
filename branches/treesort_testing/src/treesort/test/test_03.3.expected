CCþBBþDDþAA
2009-10-02þ8HPzHGZLDfþ43.1227þ222
2009-10-02þMP3BenV_PWþ19.8086þ525
2009-10-02þS-BUVAIarsþ23.1071þ120
2009-10-02þcEXl50cdsGþ23.1071þ790
2009-10-02þcOA-GflWf9þ19.8086þ120
2009-10-02þvT9F16VWxuþ16.2961þ876
2009-10-06þPKndexdB5Yþ43.1227þ119
2009-10-06þRHEIKKC-eNþ16.2961þ876
2009-10-06þVjx-O1uUiyþ23.1071þ119
2009-10-06þbWuILAzu5Lþ23.1071þ379
2009-10-06þw-DsqhmkP4þ16.2961þ428
2009-10-09þDMqu6KjLMnþ21.4784þ428
2009-10-09þNrNOPtKMh2þ23.1071þ222
2009-10-09þOcgRMuIUVWþ25.4415þ120
2009-10-09þV_Srbg1Xgkþ19.8086þ525
2009-10-09þZ_RNo9lPThþ21.4784þ222
2009-10-09þZeP1N6ZHo8þ11.9550þ222
2009-10-09þ_6GXpFrx7Mþ21.4784þ704
2009-10-09þl0sIDKi1Mnþ43.1227þ222
2009-10-15þ-_WgFpelwTþ21.4784þ119
2009-10-15þ1aRiJ5zvaUþ14.6088þ379
2009-10-15þOxNCXf_QO5þ11.9550þ222
2009-10-15þOypnZVbJOYþ13.3454þ876
2009-10-15þQ99ceOGe58þ14.6088þ119
2009-10-15þU_DsBKeZBOþ16.2961þ790
2009-10-15þV-dWH8XhzNþ11.9550þ704
2009-10-15þbNfKeDJp35þ13.3454þ120
2009-10-15þccfVWBl1Ifþ25.4415þ379
2009-10-15þpUbcEzSLg5þ43.1227þ379
2009-10-15þqHbD7IgvBsþ19.8086þ119
2009-10-15þtH_urSAL6Iþ20.0623þ379
2009-10-18þ-ynhLNQjXnþ43.1227þ525
2009-10-18þ3Abgi5FvpKþ16.2961þ692
2009-10-18þFJFwWbUCEgþ16.2961þ525
2009-10-18þQOhQYF2L4Uþ20.0623þ120
2009-10-18þR0lHfvqeE2þ20.0623þ222
2009-10-18þY0_LegXodZþ14.6088þ120
2009-10-18þczHP0Y09gOþ25.4415þ119
2009-10-18þdCzebYQfiHþ13.3454þ222
2009-10-18þi0-lOQWHXeþ23.1071þ428
2009-10-18þmdg21XYvE4þ13.3454þ379
2009-10-18þoj-npio3Cqþ11.9550þ119
2009-10-18þqv6ulolvVmþ14.6088þ876
2009-10-19þ-ca4VazCCoþ43.1227þ525
2009-10-19þ-l43yW3QQXþ21.4784þ379
2009-10-19þ0J2H9eMxz5þ19.8086þ428
2009-10-19þ1Qbw9NbEK2þ19.8086þ120
2009-10-19þ8ZDolJWFhKþ25.4415þ120
2009-10-19þDx5ojQn-G6þ13.3454þ379
2009-10-19þNqFghFb16Pþ25.4415þ119
2009-10-19þQHJtym0emOþ14.6088þ790
2009-10-19þQOa1zDDXrcþ21.4784þ692
2009-10-19þQddzqzZhvaþ20.0623þ222
2009-10-19þZs9uH9wqwvþ21.4784þ120
2009-10-19þ_okpFO60P5þ21.4784þ120
2009-10-19þnLKFOKtBt5þ21.4784þ876
2009-10-19þpPrA9-ibvPþ23.1071þ790
2009-10-19þs5FX4ERZP8þ20.0623þ119
2009-10-19þuGBjnSxQwIþ23.1071þ692
2009-10-19þyj8FAR6sT8þ21.4784þ790
2009-10-21þ0CjQNzoprAþ25.4415þ790
2009-10-21þBuPJhoNfZYþ13.3454þ692
2009-10-21þD6DavOvkp5þ13.3454þ119
2009-10-21þMk8GiTvT-2þ19.8086þ222
2009-10-21þPb0jUNXPGFþ11.9550þ704
2009-10-21þQEIlvFBOAEþ14.6088þ428
2009-10-21þUH3YCtEsnEþ20.0623þ379
2009-10-21þZ971TGdirGþ23.1071þ876
2009-10-21þj-OdwoZTzmþ23.1071þ692
2009-10-21þlAkjpi7B0uþ11.9550þ704
2009-10-21þooYbIsNctcþ14.6088þ692
2009-10-21þqdigsUU1EHþ11.9550þ222
2009-10-21þzHaJcq1brbþ43.1227þ379
2009-10-23þ6RIZ_xP77Oþ21.4784þ876
2009-10-23þHF90x6_QiBþ13.3454þ222
2009-10-23þQg_963dGw2þ25.4415þ428
2009-10-23þWxgda_Rz1uþ19.8086þ120
2009-10-23þab_ALFxdBlþ16.2961þ704
2009-10-23þgplXXJQCbDþ20.0623þ692
2009-10-25þGkijsWrtDmþ16.2961þ222
2009-10-25þK26asHqM1Bþ21.4784þ692
2009-10-25þMLySuT_6L-þ21.4784þ119
2009-10-25þN1CuDuqLsqþ19.8086þ120
2009-10-25þSEiwhaVJcvþ20.0623þ790
2009-10-25þgbwjbbmOR5þ13.3454þ704
2009-10-25þghqsjQH7CAþ21.4784þ790
2009-10-25þi4Vn-mjUX_þ20.0623þ379
2009-10-25þkC3oWoj1_8þ21.4784þ692
2009-10-25þn1pETdd_e2þ25.4415þ790
2009-10-29þ5-dmH-yGymþ11.9550þ120
2009-10-29þAnKSv96_ykþ16.2961þ525
2009-10-29þCwEfrzVzlNþ19.8086þ704
2009-10-29þH3laCVXVqFþ20.0623þ525
2009-10-29þJw-SmuJJwHþ16.2961þ428
2009-10-29þXdQde3c2LRþ16.2961þ120
2009-10-29þ_Dh-EwmwHUþ13.3454þ222
2009-10-29þ_mNRE6q64Dþ23.1071þ525
2009-10-29þeGWkG9dnHHþ13.3454þ790
2009-10-29þkX69CHRCpFþ16.2961þ119
2009-10-29þpfetgY4yYtþ21.4784þ525
